File: services/core-svc/src/main/java/com/supportme/pay/coresvc/initcode/InitCodeRepository.java
package com.supportme.pay.coresvc.initcode;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface InitCodeRepository extends JpaRepository<InitCode, Long> {
    Optional<InitCode> findByUuidAndActiveTrue(String uuid);

    List<InitCode> findByOrganizationId(Long organizationId);

    List<InitCode> findByOwnerUserId(Long ownerUserId);
}

File: services/core-svc/src/main/kotlin/com/supportme/pay/coresvc/initcode/InitCodeRepository.kt
package com.supportme.pay.coresvc.initcode

import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface InitCodeRepository : JpaRepository<InitCode, Long> {
    fun findByUuidAndActiveTrue(uuid: String): Optional<InitCode>
    fun findByOrganizationId(organizationId: Long): List<InitCode>
    fun findByOwnerUserId(ownerUserId: Long): List<InitCode>
}
